README.md

```diff
@@ -8,3 +8,157 @@
+
+## Usage
+
+### Setup
+
+Initialize the VelocidiSDK with the necessary `trackingHost` and the `matchHost` URLs. Without this, VelocidiSDK will not work. We suggest doing this when the application launches.
+
+__Swift__
+```swift
+import VelocidiSDK
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+    var window: UIWindow?
+
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        // Override point for customization after application launch.
+
+        let config = VSDKConfig(hosts: "https://tr.yourdomain.com", "https://match.yourdomain.com")!
+        VSDKVelocidi.start(config)
+        return true
+    }
+```
+
+__Objective-C__
+```objc
+@import VelocidiSDK;
+
+@interface AppDelegate ()
+
+@end
+
+@implementation AppDelegate
+
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    // Override point for customization after application launch.
+    
+    VSDKConfig * config = [[VSDKConfig alloc] initWithHosts:@"https://tr.yourdomain.com": @"https://match.yourdomain.com"];
+    [VSDKVelocidi start: config];
+    return YES;
+}
+```
+
+### Send a track event
+
+In order to sent a tracking event, create an instance of TrackingEvent. Then, call the singleton instance of `VSDKVelocidi` and use the method `track`
+
+__Swift__
+```swift
+import VelocidiSDK
+
+...
+
+let trackingEvent = VSDKPageView()
+trackingEvent.siteId = "RandomSiteId"
+trackingEvent.clientId = "RandomClientId"
+
+VSDKVelocidi.sharedInstance().track(trackingEvent)
+```
+
+__Objective-C__
+```objc
+@import VelocidiSDK;
+
+...
+
+VSDKTrackingEvent * trackingEvent =  [[VSDKPageView alloc] init];
+trackingEvent.clientId = @"RandomSiteId";
+trackingEvent.siteId = @"RandomClientId";
+
+[VSDKVelocidi.sharedInstance track: trackingEvent] 
+```
+
+
+You can also pass callback blocks that will that will be called when the request is successful or if if fails.
+
+__Swift__
+```swift
+VSDKVelocidi.sharedInstance().track(trackingEvent, onSuccess:{ (response: URLResponse, responseObject: Any) in
+  print("Success! Response: \(response)")
+}, onFailure:{(error: Error) in
+  print("Failed! Error: \(error.localizedDescription)")
+})
+```
+
+__Objective-C__
+```objc
+[VSDKVelocidi.sharedInstance track: trackingEvent onSuccess: ^(NSURLResponse * response, id responseObject){
+    NSLog(@"Success! Response: %@", trackingNumber);
+} onFailure: ^(NSError * error){
+    NSLog(@"Failed! Error: %@", [error localizedDescription]);
+}];
+```
+
+#### Available tracking events model classes
+* VSDKAddToCart
+* VSDKPageView
+* VSDKProductClick
+* VSDKProductCustomization
+* VSDKProductFeedback
+* VSDKProductImpression
+* VSDKProductView
+* VSDKProductViewDetails
+* VSDKPurchase
+* VSDKRefund
+* VSDKRemoveFromCart
+* VSDKSearch
+* VSDKSubscription
+
+<!-- #### Create your custom tracking event -->
```

Comment:
Either commented or deleted. Ran into troubles when doing this. We cannot use JSONModel, the framework we use to serialize classes to JSON, in Swift and therefore we cannot make custom tracking events.

Reply:
What alternatives a developer has to send a custom event?

Reply:
If he uses Objective-C, inherit `VSDKTrackingEvent` or any other class that also inherits from `VSDKTrackingEvent`. If he uses Swift, none.

Reply:
Can't we implement a new method that receives a dictionary and sends it as JSON?

Reply:
Maybe, I'd have to look at it. But ideally, we would not have to move away from inheritance or from using objects. Either way, I just left this here because I ended up forgetting I had commented it (this does not appear when the .md is rendered so I forgot that I had written it). Meant to leave this for a proper discussion about what we should do:
* Document how to use custom events in Objective-C and don't offer this functionality in Swift.
* R&D how to properly fix this, maybe by changing to a new JSON library.
* Avoid the problem by doing something like what you suggested.

Reply:
If I have a Swift project, can I write an Objective-C class and use it in Swift?

Reply:
Should work, yes.

Reply:
At least for now, that seems good enough for me. What do you think? Can you document it?

Reply:
Yes, I'll look into it. It's not a very good approach and we should revise this in the future, but I think that for now is the best we have.

Reply:
I noticed you re-requested my review. Has this been fixed?

Reply:
Sorry, my bad 😅

Reply:
Please take a look at 9bf9e78 d201da1 18be956 and d394ec1